src/validator.rs

```diff
@@ -19,10 +19,32 @@ pub struct DefaultValidator;
 
 impl Validator for DefaultValidator {
     fn validate(&self, line: &str) -> ValidationResult {
-        if line.split('"').count() % 2 == 0 {
+        if line.split('"').count() % 2 == 0 || incomplete_brackets(line) {
             ValidationResult::Incomplete
         } else {
             ValidationResult::Complete
         }
     }
 }
+
+fn incomplete_brackets(line: &str) -> bool {
+    let mut balance: Vec<char> = Vec::new();
+
+    for c in line.chars() {
+        if c == '{' {
+            balance.push('}')
+        } else if c == '[' {
+            balance.push(']')
+        } else if c == '(' {
+            balance.push(')')
```

Comment:
how about -
```rust
if ['{', '[', '('].contains(&c) {
   balance.push(c)
else if ['}', ']', ')'].contains(&c) {
   ...
}
```
I just find that the if and else blocks look similar (complimentary?) in this one.

Reply:
They are not similar. If you look at the cases, each insert the opposite pair to the condition. Probably it would be a good idea to use a enum and move those cases to the implementations of the enum. However, I wanted to keep the default validator simple and very explicit